File: Python/basketball_travel_fares/travel_fares.py
import math

fares =  {  "Oromocto": {"fare": 15, "distance" : 21.5},
            "Base Gagetown": {"fare": 15, "distance": 25.8},
            "Keswick Ridge": {"fare": 19, "distance": 23.2},
            "Keswick Valley": {"fare": 22, "distance": 29.5},
            "Fredericton Junction": {"fare": 30, "distance": 36.6},
            "Harvey": {"fare": 35, "distance": 49.2},
            "Stanley": {"fare": 38, "distance": 42.6},
            "Minto": {"fare": 38, "distance": 65.3},
            "Gagetown": {"fare": 45, "distance": 58.2},
            "Nackawic": {"fare": 48, "distance": 63.9},
            "Cambridge Narrows": {"fare": 53, "distance": 68.8},
            "Chipman": {"fare": 57, "distance": 91.7},
            "Central NB Academy": {"fare": 60, "distance": 76.9}
                                
}
            
for location, fare_data in fares.items() :
    p = fare_data["fare"]
    d = fare_data["distance"]
    p_km = p / d
    l = len(location)
    for i in range(min(21, l), 21) :
        location += " "
    print("\t$" + str(p) + " for going to " + location + " " + str(d) + " km, ratio: " + str(p_km) + " $ / km.")

File: Java/BasketBallRefereeing/src/sample/TravelFares.java
package sample;

import java.util.Comparator;

public enum TravelFares {
    OUT_OF_ZONE("Out of zone", 0,0),
    FREDERICTON("Fredericton",0, 0),
    STANLEY("Stanley",38,42.6),
    CAMBRIDGE_NARROWS("Cambridge Narrows",53,68.8),
    BOIESTOWN("Boiestown",60,76.9),
    HARVEY("Harvey",35,49.2),
    OROMOCTO("Oromocto",15,21.5),
    BASE_GAGETOWN("Base Gagetown", 15, 25.8),
    GAGETOWN("Gagetown",45,58.2),
    KESWICK_VALLEY("Keswick Valley",22,29.5),
    KESWICK_RIDGE("Keswick Ridge",19,23.2),
    CHIPMAN("Chipman",57,91.7),
    MINTO("Minto",38,65.3),
    MCADAM("McAdam",60,80.7),
    NACKAWICK("Nackawick",48,63.9),
    FREDERICTON_JUNCTION("Fredericton Junction",30,36.6);
    private String locationName;
    private double travelFare, distance;

    TravelFares(String locationName, double travelFare, double distance) {
        this.locationName = locationName;
        this.travelFare = travelFare;
        this.distance = distance;
    }

    public String getLocationName() {
        return this.locationName;
    }

    public double getDollarPerKM() {
        return this.travelFare / this.distance;
    }

    public static Comparator<TravelFares> compareTravelFareNames() {
        return Comparator.comparing(TravelFares::getLocationName);
    }

    public String toString() {
        return locationName + ", distance: " + distance + ", travel fare: $ " + travelFare + ".";
    }
}
